pkg/cache/cache_test.go

```diff
@@ -2549,141 +2518,93 @@ func TestCacheQueueOperations(t *testing.T) {
 			key:                "ns1/alpha",
 			reservingWorkloads: 1,
 			admittedWorkloads:  1,
-			usage: FlavorResourceQuantities{
-				"spot": {
-					corev1.ResourceCPU:    workload.ResourceValue(corev1.ResourceCPU, resource.MustParse("2")),
-					corev1.ResourceMemory: workload.ResourceValue(corev1.ResourceMemory, resource.MustParse("8Gi")),
-				},
-				"model-a": {
-					"example.com/gpu": workload.ResourceValue("example.com/gpu", resource.MustParse("0")),
-				},
-			},
-			admittedUsage: FlavorResourceQuantities{
-				"spot": {
-					corev1.ResourceCPU:    workload.ResourceValue(corev1.ResourceCPU, resource.MustParse("2")),
-					corev1.ResourceMemory: workload.ResourceValue(corev1.ResourceMemory, resource.MustParse("8Gi")),
-				},
-				"model-a": {
-					"example.com/gpu": workload.ResourceValue("example.com/gpu", resource.MustParse("0")),
-				},
-			},
+			usage: resources.FlavorResourceQuantitiesFlat{
+				resources.FlavorResource{Flavor: "spot", Resource: corev1.ResourceCPU}:    workload.ResourceValue(corev1.ResourceCPU, resource.MustParse("2")),
+				resources.FlavorResource{Flavor: "spot", Resource: corev1.ResourceMemory}: workload.ResourceValue(corev1.ResourceMemory, resource.MustParse("8Gi")),
+				resources.FlavorResource{Flavor: "model-a", Resource: "example.com/gpu"}:  workload.ResourceValue("example.com/gpu", resource.MustParse("0")),
```

Comment:
I am wondering if it won't be better to unify the test cases, to directly pass the resource values?

Like here:
```suggestion
				resources.FlavorResource{Flavor: "spot", Resource: corev1.ResourceCPU}: 2_000,
				resources.FlavorResource{Flavor: "spot", Resource: corev1.ResourceMemory}: 8 * utiltesting.Gi,
				resources.FlavorResource{Flavor: "model-a", Resource: "example.com/gpu"}:  0,
```

Reply:
Perhaps it is worth doing that, but it is outside the scope of this PR. Additionally, I wanted to avoid any logic changes, regardless of how trivial, given the scope of the changes.